token/services/ttx/info.go

```diff
@@ -46,37 +46,39 @@ func (a *TransactionInfoProvider) TransactionInfo(txID string) (*TransactionInfo
 		return nil, errors.WithMessagef(err, "failed to load endorsement acks for [%s]", txID)
 	}
 
-	applicationMetadata, err := a.loadTransient(txID)
-	if err != nil {
-		return nil, errors.WithMessagef(err, "failed to load transient for [%s]", txID)
-	}
-
 	tr, err := a.ttxDB.GetTokenRequest(txID)
 	if err != nil {
 		return nil, errors.WithMessagef(err, "failed to load token request for [%s]", txID)
 	}
 
+	applicationMetadata, err := a.loadTransient(tr, txID)
+	if err != nil {
+		return nil, errors.WithMessagef(err, "failed to load transient for [%s]", txID)
+	}
 	return &TransactionInfo{
 		EndorsementAcks:     endorsementAcks,
 		ApplicationMetadata: applicationMetadata,
 		TokenRequest:        tr,
 	}, nil
 }
 
-func (a *TransactionInfoProvider) loadTransient(txID string) (map[string][]byte, error) {
-	tm, err := network.GetInstance(a.sp, a.tms.Network(), a.tms.Channel()).GetTransient(txID)
-	if err != nil {
-		return nil, errors.WithMessagef(err, "failed to load transient for [%s]", txID)
-	}
-	if !tm.Exists(TokenRequestMetadata) {
+func (a *TransactionInfoProvider) loadTransient(trRaw []byte, txID string) (map[string][]byte, error) {
+	if len(trRaw) == 0 {
+		if logger.IsEnabledFor(zapcore.DebugLevel) {
```

Comment:
You don't have to make the check isEnabledFor unless you have any calculations to do.